pkg/runner/entry.go

```diff
@@ -0,0 +1,199 @@
+package runner
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"sort"
+
+	"github.com/golang-collections/collections/set"
+	"kcl-lang.io/kpm/pkg/constants"
+	"kcl-lang.io/kpm/pkg/errors"
+	"kcl-lang.io/kpm/pkg/reporter"
+	"kcl-lang.io/kpm/pkg/utils"
+)
+
+// Entry is the entry of 'kpm run'.
+type Entry struct {
+	// The package source of the entry, filepath, tar path, url or ref.
+	packageSource string
+	// The start files for one compilation.
+	entryFiles []string
+	// The kind of the entry, file, tar, url or ref.
+	kind string
```

Comment:
Use a typedef `EntryKind` for the `kind` field.

Besides, the concept of fakePackage is somewhat difficult to understand. If it is one of the ways to enter, I suggest using it as an option for the `kind` field.

Reply:
fixed.